src/audio/google/google_rtc_audio_processing.c

```diff
@@ -93,6 +108,130 @@ void GoogleRtcFree(void *ptr)
 	return rfree(ptr);
 }
 
+static ALWAYS_INLINE float clamp_rescale(float max_val, float x)
+{
+	float min = -1.0f;
```

Comment:
I suggest using HiFi coprocessor for this (peformance!!)

Reply:
The compiler automatically emits HiFi instructions here (or FPU, for example on platforms that have HiFi3 w/o floating point support).   There was some disassembly posted months ago in a previous iteration of this PR that showed what the inner loops look like.  Basically the resulting code here is pretty much optimal.  Actually if you build with -O3 (we don't) xt-clang is capable of autovectorizing the loop for another ~2x benefit on HiFi4/5 platforms.